File: Week 1/Credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long credit_card_number = get_long_long("Number: ");

    int card_numbers_to_check_sum = 0;
    long credit_card_number_to_check = credit_card_number / 10;

    while (credit_card_number_to_check > 0)
    {
        int number = credit_card_number_to_check % 10 * 2;
        if (number < 10)
        {
            card_numbers_to_check_sum += number;
        }
        else
        {
            card_numbers_to_check_sum += number % 10;
            number /= 10;
            card_numbers_to_check_sum += number;
        }

        credit_card_number_to_check /= 100;
    }

    credit_card_number_to_check = credit_card_number;
    while (credit_card_number_to_check > 0)
    {
        card_numbers_to_check_sum += credit_card_number_to_check % 10;
        credit_card_number_to_check /= 100;
    }

    if (card_numbers_to_check_sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        int length = 2;
        while (credit_card_number >= 100)
        {
            credit_card_number /= 10;
            length++;
        }

        if ((credit_card_number == 34 || credit_card_number == 37) && length == 15)
        {
            printf("AMEX\n");
        }
        else if (credit_card_number >= 51 && credit_card_number <= 55 && length == 16)
        {
            printf("MASTERCARD\n");
        }
        else if (credit_card_number >= 40 && credit_card_number <= 49 && (length == 13 || length == 16))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
}

File: Week 6/sentimental-credit/credit.py
from cs50 import *


def main():
    number = cs50.get_string("Number: ")
    number_lenght = len(number)

    if (number_lenght != 13 and number_lenght != 15 and number_lenght != 16):
        print("INVALID")
        return

    sum_numbers = 0
    i = number_lenght - 1
    while (i >= 0):
        sum_numbers += int(number[i])
        temp_sum_multiplied_numbers = int(number[i-1]) * 2
        if (i - 1 >= 0):
            if (temp_sum_multiplied_numbers < 10):
                sum_numbers += temp_sum_multiplied_numbers
            else:
                temp_sum_multiplied_numbers_str = str(temp_sum_multiplied_numbers)
                sum_numbers += int(temp_sum_multiplied_numbers_str[0])
                sum_numbers += int(temp_sum_multiplied_numbers_str[1])
        i -= 2

    if (sum_numbers % 10 != 0):
        print("INVALID")
        return

    if ((int(number[:2]) == 34 or int(number[:2]) == 37) and number_lenght == 15):
        print("AMEX")
    elif ((int(number[:2]) >= 51 and int(number[:2]) <= 55) and number_lenght == 16):
        print("MASTERCARD")
    elif ((int(number[:2]) >= 40 and int(number[:2]) <= 49) and (number_lenght == 13 or number_lenght == 16)):
        print("VISA")
    else:
        print("INVALID")
        return


main()
